src/transformStateWithClones.js

```diff
@@ -7,7 +8,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let newObject = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      newObject = { ...newObject, ...extraData };
+    }
+
+    if (type === 'removeProperties') {
+      for (const item of keysToRemove) {
+        delete newObject[item];
+      }
+    }
+
+    if (type === 'clear') {
+      for (const key in newObject) {
+        delete newObject[key];
+      }
+    }
```

Comment:
Instead of multiple 'if' statements, use a 'switch' statement to handle different action types since you have a limited and known set of action types. This will make the code cleaner and more maintainable.